src/hotspot/share/logging/logFileOutput.cpp

```diff
@@ -195,7 +195,16 @@ bool LogFileOutput::parse_options(const char* options, outputStream* errstream)
     char* value_str = equals_pos + 1;
     *equals_pos = '\0';
 
-    if (strcmp(FileCountOptionKey, key) == 0) {
+    if (strcmp(FoldMultilinesOptionKey, key) == 0) {
+      // We need to pass <key>=<value> style option to LogFileStreamOutput::initialize().
+      // Thus we restore '=' temporally.
+      *equals_pos = '=';
+      success = LogFileStreamOutput::initialize(pos, errstream);
+      *equals_pos = '\0';
+      if (!success) {
```

Comment:
I think the code can be simplified by parsing the value here:
```
bool foldmultilines = ....;
LogFileStreamOutput::set_foldmultilines(foldmultilines);
```

Reply:
`foldmultilines` is a parameter which should be applied in initialization. `LogOutput` and extended classes of it look like to have a rule to apply initialization parameter at `initialize()`.
We can simplify the code as you said if we can apply `foldmultiline` for `LogFileStreamOutput` at `LogFileOutput::initialize()`, but I concern it breaks the semantics for UL implementation.

Reply:
If I understand correctly, your concern is `_fold_multilines` is a property of `LogFileStreamOutput`, so the parsing of this option should be done inside the `LogFileStreamOutput` class. That way, (in the future) all subclasses of `LogFileStreamOutput` will be able to handle `foldmultiline=true` in their options by deferring to `LogFileStreamOutput::initialize`.

I suppose this would be useful if we have support for sockets in the future, like

```
-Xlog:all=socket::address=10.1.1.1,port=1234,foldmultiline=true
```

However, I don't think your current implementation has the right abstraction -- `FoldMultilinesOptionKey` is now checked both in `LogFileStreamOutput::initialize()` and `LogFileOutput::initialize()`. If we add a new `LogSocketOutput` in the future, we would have to duplicated the code in `LogSocketOutput::initialize()` as well.

I think we should defer the design of such an abstraction until we need it. Ideally, `LogFileOutput` and `LogSocketOutput` should not need to know what options are supported by their superclass, `LogFileStreamOutput`

 For the time being, it's easier to parse all the file output options in `LogFileOutput::initialize()` to keep the code simple.

Reply:
> I don't think your current implementation has the right abstraction

Agree, I think we can (should) improve more. Its improvement is not for `LogSocketStream` but also for `LogStd{out,err}Output`.
As we discussed in CSR about stdout/err, we will work for them in next step. But they extend `LogFileStreamOutput` directly. So I want to refactor it before next step.

Anyway I moved `foldmultiline` handling into `LogFileOutput` in new commit.